File: solutions/go/binary-search/1/binary_search.go
package binarysearch

func SearchInts(input []int, key int) int {
	return binarySearch(input, key, 0, len(input)-1)
}

func binarySearch(arr []int, key int, left int, right int) int {
	if left > right {
		return -1
	}
	mid := (left + right) / 2
	switch {
	case arr[mid] == key:
		return mid
	case arr[mid] < key:
		return binarySearch(arr, key, mid+1, right)
	default: // arr[mid] > key
		return binarySearch(arr, key, left, mid-1)
	}
}

File: solutions/python/binary-search/1/binary_search.py
def find(search_list, value):
    def binary_search(arr, low, high, value):
        if high >= low: 
            mid = (high + low) // 2 
            if arr[mid] == value:
                return mid
            elif arr[mid] > value:
                return binary_search(arr, low, mid - 1, value)
            else:
                return binary_search(arr, mid + 1, high, value) 
        else:
            raise ValueError("value not in array")
    return binary_search(search_list, 0, len(search_list)-1, value)
